### PR TITLE
Add global.prometheusUrl field for byop use-case

### PR DESCRIPTION
**Pull request by Pothulapati**
Implements third part of linkerd/rfc#16, #4375

This pr adds `globa.prometheusUrl` which will be used to configure publlic-api, hearbeat, grafana to use existing prometheus on the query path!

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

File: charts/linkerd2/templates/heartbeat.yaml [hunk 1]
```diff
@@ -39,9 +39,13 @@ spec:
             imagePullPolicy: {{.Values.global.imagePullPolicy}}
             args:
             - "heartbeat"
-            - "-prometheus-url=http://linkerd-prometheus.{{.Values.global.namespace}}.svc.{{.Values.global.clusterDomain}}:9090"
             - "-controller-namespace={{.Values.global.namespace}}"
             - "-log-level={{.Values.controllerLogLevel}}"
+            {{- if .Values.global.prometheusUrl }}
+            - "-prometheus-url={{.Values.global.prometheusUrl}}"
+            {{- else if .Values.prometheus.enabled }}
```
Comment (alpeb) [inline charts/linkerd2/templates/heartbeat.yaml:46]: Seems you'll have to leave this at just `else` for now for the integration tests to pass

File: charts/add-ons/grafana/templates/grafana.yaml [hunk 2]
```diff
@@ -43,7 +43,11 @@ data:
       type: prometheus
       access: proxy
       orgId: 1
+      {{- if .Values.global.prometheusUrl }}
+      url: {{.Values.global.prometheusUrl}}
+      {{- else if .Values.prometheus.enabled }}
```
Comment (alpeb) [inline charts/add-ons/grafana/templates/grafana.yaml:48]: same here

File: charts/linkerd2/templates/controller.yaml [hunk 3]
```diff
@@ -69,7 +69,11 @@ spec:
       containers:
       - args:
         - public-api
-        - -prometheus-url=http://linkerd-prometheus.{{.Values.global.namespace}}.svc.{{.Values.global.clusterDomain}}:9090
+        {{- if .Values.global.prometheusUrl }}
+        - -prometheus-url={{.Values.global.prometheusUrl}}
+        {{- else if .Values.prometheus.enabled }}
```
Comment (alpeb) [inline charts/linkerd2/templates/controller.yaml:74]: and here

**Comment by grampelberg** [comment]
@Pothulapati coming back to this one?

**Comment by Pothulapati** [comment]
Yep. That sounds right 👍

**Comment by Pothulapati** [comment]
Okay, This was working well but suddently me and @alpeb have been caught up in a weird case, where `public-api` complains
```
⛵ kind-kind in linkerd2 on  tarun/byop [$?] via 🐹 v1.14.6 via ⬢ v10.19.0
❯ k -n linkerd logs deploy/linkerd-controller public-api
WARNING: Package "github.com/golang/protobuf/protoc-gen-go/generator" is deprecated.
        A future release of golang/protobuf will delete this package,
        which has long been excluded from the compatibility promise.

time="2020-07-23T16:03:01Z" level=info msg="running version git-8f481bb7"
time="2020-07-23T16:03:01Z" level=info msg="Using cluster domain: cluster.local"
time="2020-07-23T16:03:01Z" level=info msg="waiting for caches to sync"
time="2020-07-23T16:03:01Z" level=info msg="caches synced"
time="2020-07-23T16:03:01Z" level=info msg="starting admin server on :9995"
time="2020-07-23T16:03:01Z" level=info msg="starting HTTP server on :8085"
time="2020-07-23T16:05:13Z" level=error msg="Query(max(process_start_time_seconds{}) by (pod, namespace)) failed with: server_error: server error: 502"
time="2020-07-23T16:07:18Z" level=error msg="Query(max(process_start_time_seconds{}) by (pod, namespace)) failed with: server_error: server error: 502"
```

Will debug more and update the PR accordingly.

**Pull request merged**